tests/rgb_tests.cpp

```diff
@@ -131,3 +134,103 @@ TEST(rgb, linear_to_srgb8_lut)
 		EXPECT_EQ(linear_to_srgb8_lut(lin), expected);
 	}
 }
+
+TEST(rgb, rgb565_in_memory_order)
+{
+	//                                  | red | green| blue|
+	constexpr uint16_t rgb_as_uint16 = 0b00001'000011'00111;
+	constexpr uint8_t r8             = 0b00001'000;
+	constexpr uint8_t g8             = /* */ 0b000011'00;
+	constexpr uint8_t b8             = /*        */ 0b00111'000;
+
+	const Rgb565 rgb_object_from_uint16(rgb_as_uint16);
+	const Rgb565 rgb_object_from_components(r8, g8, b8);
+
+	EXPECT_EQ(rgb_object_from_uint16.pixel, rgb_object_from_components.pixel);
+}
+
+TEST(rgb, rgb666_in_memory_order)
+{
+	constexpr uint8_t r6 = 0b100'011;
+	constexpr uint8_t g6 = 0b100'111;
+	constexpr uint8_t b6 = 0b101'111;
+
+	constexpr Rgb666 rgb_object(r6, g6, b6);
+
+	constexpr auto array_size = sizeof(rgb_object);
+
+	// Copy the object byte-by-byte into an array
+	uint8_t rgb_array[array_size] = {};
+	memcpy(rgb_array, reinterpret_cast<const void*>(&rgb_object), array_size);
+
+	// Do the array elements match the RGB values we put in?
+	EXPECT_EQ(rgb_array[0], r6);
+	EXPECT_EQ(rgb_array[1], g6);
+	EXPECT_EQ(rgb_array[2], b6);
+}
+
+TEST(rgb, rgb888_in_memory_order)
+{
+	constexpr uint8_t r8 = 0b1000'0011;
+	constexpr uint8_t g8 = 0b1000'0111;
+	constexpr uint8_t b8 = 0b1000'1111;
+
+	constexpr Rgb888 bgra_object(r8, g8, b8);
+
+	// Cast the object to a uint32_t
+	constexpr uint32_t rgb_as_uint32 = bgra_object;
+
+	constexpr auto array_size = sizeof(rgb_as_uint32);
+
+	// Copy it byte-by-byte into an array
+	uint8_t rgb_array[array_size] = {};
+	memcpy(rgb_array, &rgb_as_uint32, sizeof(rgb_as_uint32));
+
+	// Do the array elements match the RGB values we put in?
+
+	// Rgb888 represents the colour value as a packed integer
+	// having endian-specific memory order or byte-order.
+#ifdef WORDS_BIGENDIAN
+
+	// On big endian, the byte-order ia ABGR
+	EXPECT_EQ(rgb_array[0], 0);
+	EXPECT_EQ(rgb_array[1], b8);
+	EXPECT_EQ(rgb_array[2], g8);
+	EXPECT_EQ(rgb_array[3], r8);
+#else
+	// On little endian, the byte-order ia RGBA
+	EXPECT_EQ(rgb_array[0], r8);
+	EXPECT_EQ(rgb_array[1], g8);
+	EXPECT_EQ(rgb_array[2], b8);
+#endif
```

Comment:
Similarly, why do we need this? We already have the `BGR24_ByteArray` pixel format. Is it really true that some parts of the code in fact use the RGB24_ByteArray format without having an actual pixel format definition for it by mucking around with casting things to `Rgb888`? If so, that's rather unfortunate...

Reply:
I've updated the unit tests to just sanity check byte-order and values (and exluded Rgb666).  

Although our DOS video-card memory is just uint8_t*, the tests confirm that our structs (Rgb888 and Bgrx8888) can be used to go to-or-from memory directly. More importantly, that the underlying representation is endian-agnostic, too.